<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>886</width>
    <height>1038</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="0" column="2">
     <widget class="QLabel" name="graphvizImg">
      <property name="text">
       <string/>
      </property>
      <property name="pixmap">
       <pixmap>C:/Users/User/Downloads/Telegram Desktop/photo_2024-06-24_08-07-17.jpg</pixmap>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QWidget" name="gridWidget_2" native="true">
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <layout class="QVBoxLayout" name="_3">
         <property name="sizeConstraint">
          <enum>QLayout::SetMaximumSize</enum>
         </property>
         <property name="leftMargin">
          <number>10</number>
         </property>
         <property name="topMargin">
          <number>10</number>
         </property>
         <property name="rightMargin">
          <number>10</number>
         </property>
         <property name="bottomMargin">
          <number>10</number>
         </property>
         <item alignment="Qt::AlignHCenter">
          <widget class="QLabel" name="label_8">
           <property name="font">
            <font>
             <family>CADT Mono Display</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Finite Automata</string>
           </property>
           <property name="margin">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget" native="true">
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="3" column="0">
             <widget class="QLabel" name="label_9">
              <property name="styleSheet">
               <string notr="true">QLabel {
                background-color: #f0f0f0; /* Light gray background */
                color: #333; /* Dark gray text color */
                padding: 10px;
                border-radius: 5px; /* Rounded corners */
                font-size: 18px; /* Font size */
                font-weight: bold; /* Bold text */
                text-align: center; /* Center text */
            }</string>
              </property>
              <property name="text">
               <string>Start State</string>
              </property>
             </widget>
            </item>
            <item row="3" column="2">
             <widget class="QLineEdit" name="startStatesLineEdit">
              <property name="styleSheet">
               <string notr="true">QLineEdit {
    background-color: #f0f0f0; /* Light gray background */
    border: 2px solid #ccc; /* Light gray border */
    padding: 5px; /* Padding inside the QLineEdit */
    color: #333; /* Dark gray text color */
    border-radius: 5px; /* Rounded corners */
}

QLineEdit:hover {
    border-color: #aaa; /* Darker border color on hover */
}

QLineEdit:focus {
    border-color: #4CAF50; /* Green border color when focused */
}
</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_10">
              <property name="styleSheet">
               <string notr="true">QLabel {
                background-color: #f0f0f0; /* Light gray background */
                color: #333; /* Dark gray text color */
                padding: 10px;
                border-radius: 5px; /* Rounded corners */
                font-size: 18px; /* Font size */
                font-weight: bold; /* Bold text */
                text-align: center; /* Center text */
            }</string>
              </property>
              <property name="text">
               <string>Transitions</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLineEdit" name="statesLineEdit">
              <property name="styleSheet">
               <string notr="true">QLineEdit {
    background-color: #f0f0f0; /* Light gray background */
    border: 2px solid #ccc; /* Light gray border */
    padding: 5px; /* Padding inside the QLineEdit */
    color: #333; /* Dark gray text color */
    border-radius: 5px; /* Rounded corners */
}

QLineEdit:hover {
    border-color: #aaa; /* Darker border color on hover */
}

QLineEdit:focus {
    border-color: #4CAF50; /* Green border color when focused */
}
</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_11">
              <property name="styleSheet">
               <string notr="true">QLabel {
                background-color: #f0f0f0; /* Light gray background */
                color: #333; /* Dark gray text color */
                padding: 10px;
                border-radius: 5px; /* Rounded corners */
                font-size: 18px; /* Font size */
                font-weight: bold; /* Bold text */
                text-align: center; /* Center text */
            }</string>
              </property>
              <property name="text">
               <string>States</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_12">
              <property name="styleSheet">
               <string notr="true">QLabel {
                background-color: #f0f0f0; /* Light gray background */
                color: #333; /* Dark gray text color */
                padding: 10px;
                border-radius: 5px; /* Rounded corners */
                font-size: 18px; /* Font size */
                font-weight: bold; /* Bold text */
                text-align: center; /* Center text */
            }</string>
              </property>
              <property name="text">
               <string>Symbols</string>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QLineEdit" name="transitionsLineEdit">
              <property name="styleSheet">
               <string notr="true">QLineEdit {
    background-color: #f0f0f0; /* Light gray background */
    border: 2px solid #ccc; /* Light gray border */
    padding: 5px; /* Padding inside the QLineEdit */
    color: #333; /* Dark gray text color */
    border-radius: 5px; /* Rounded corners */
}

QLineEdit:hover {
    border-color: #aaa; /* Darker border color on hover */
}

QLineEdit:focus {
    border-color: #4CAF50; /* Green border color when focused */
}
</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLineEdit" name="symbolsLineEdit">
              <property name="styleSheet">
               <string notr="true">QLineEdit {
    background-color: #f0f0f0; /* Light gray background */
    border: 2px solid #ccc; /* Light gray border */
    padding: 5px; /* Padding inside the QLineEdit */
    color: #333; /* Dark gray text color */
    border-radius: 5px; /* Rounded corners */
}

QLineEdit:hover {
    border-color: #aaa; /* Darker border color on hover */
}

QLineEdit:focus {
    border-color: #4CAF50; /* Green border color when focused */
}
</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_13">
              <property name="styleSheet">
               <string notr="true">QLabel {
                background-color: #f0f0f0; /* Light gray background */
                color: #333; /* Dark gray text color */
                padding: 10px;
                border-radius: 5px; /* Rounded corners */
                font-size: 18px; /* Font size */
                font-weight: bold; /* Bold text */
                text-align: center; /* Center text */
            }</string>
              </property>
              <property name="text">
               <string>Final State</string>
              </property>
             </widget>
            </item>
            <item row="4" column="2">
             <widget class="QLineEdit" name="finalStates">
              <property name="styleSheet">
               <string notr="true">QLineEdit {
    background-color: #f0f0f0; /* Light gray background */
    border: 2px solid #ccc; /* Light gray border */
    padding: 5px; /* Padding inside the QLineEdit */
    color: #333; /* Dark gray text color */
    border-radius: 5px; /* Rounded corners */
}

QLineEdit:hover {
    border-color: #aaa; /* Darker border color on hover */
}

QLineEdit:focus {
    border-color: #4CAF50; /* Green border color when focused */
}
</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QPushButton" name="clearTextBtn">
             <property name="styleSheet">
              <string notr="true">QPushButton {
                background-color: #1f6390;
                border: none;
                color: white;
                padding: 8px 16px; /* Adjusted padding */
                text-align: center;
                text-decoration: none;
                font-size: 14px; /* Adjusted font size */
                border-radius: 15px; /* Rounded corners */
                outline: none; /* No outline */
            }
            
            QPushButton:hover {
                background-color: #45a049; /* Darker green on hover */
            }
            
            QPushButton:pressed {
                background-color: #1f6390; /* Dark green when pressed */
            }</string>
             </property>
             <property name="text">
              <string>Clear</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="submitBtn">
             <property name="styleSheet">
              <string notr="true">QPushButton {
                background-color: #1f6390;
                border: none;
                color: white;
                padding: 8px 16px; /* Adjusted padding */
                text-align: center;
                text-decoration: none;
                font-size: 14px; /* Adjusted font size */
                border-radius: 15px; /* Rounded corners */
                outline: none; /* No outline */
            }
            
            QPushButton:hover {
                background-color: #45a049; /* Darker green on hover */
            }
            
            QPushButton:pressed {
                background-color: #1f6390; /* Dark green when pressed */
            }</string>
             </property>
             <property name="text">
              <string>OK</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item alignment="Qt::AlignHCenter">
            <widget class="QLabel" name="label_14">
             <property name="font">
              <font>
               <family>CADT Mono Display</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Functions</string>
             </property>
             <property name="margin">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="checkDeterministicBtn">
             <property name="palette">
              <palette>
               <active>
                <colorrole role="WindowText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Button">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>31</red>
                   <green>99</green>
                   <blue>144</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Text">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="ButtonText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Base">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>31</red>
                   <green>99</green>
                   <blue>144</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Window">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>31</red>
                   <green>99</green>
                   <blue>144</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="PlaceholderText">
                 <brush brushstyle="NoBrush">
                  <color alpha="128">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
               </active>
               <inactive>
                <colorrole role="WindowText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Button">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>31</red>
                   <green>99</green>
                   <blue>144</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Text">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="ButtonText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Base">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>31</red>
                   <green>99</green>
                   <blue>144</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Window">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>31</red>
                   <green>99</green>
                   <blue>144</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="PlaceholderText">
                 <brush brushstyle="NoBrush">
                  <color alpha="128">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
               </inactive>
               <disabled>
                <colorrole role="WindowText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Button">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>31</red>
                   <green>99</green>
                   <blue>144</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Text">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="ButtonText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Base">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>31</red>
                   <green>99</green>
                   <blue>144</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Window">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>31</red>
                   <green>99</green>
                   <blue>144</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="PlaceholderText">
                 <brush brushstyle="NoBrush">
                  <color alpha="128">
                   <red>255</red>
                   <green>255</green>
                   <blue>255</blue>
                  </color>
                 </brush>
                </colorrole>
               </disabled>
              </palette>
             </property>
             <property name="font">
              <font>
               <pointsize>-1</pointsize>
               <weight>50</weight>
               <italic>false</italic>
               <bold>false</bold>
               <underline>false</underline>
               <strikeout>false</strikeout>
              </font>
             </property>
             <property name="autoFillBackground">
              <bool>false</bool>
             </property>
             <property name="styleSheet">
              <string notr="true">QPushButton {
                background-color: #1f6390;
                border: none;
                color: white;
                padding: 8px 16px; /* Adjusted padding */
                text-align: center;
                text-decoration: none;
                font-size: 14px; /* Adjusted font size */
                border-radius: 15px; /* Rounded corners */
                outline: none; /* No outline */
            }
            
            QPushButton:hover {
                background-color: #45a049; /* Darker green on hover */
            }
            
            QPushButton:pressed {
                background-color: #1f6390; /* Dark green when pressed */
            }</string>
             </property>
             <property name="text">
              <string>Check Deterministic</string>
             </property>
             <property name="autoDefault">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="checkStringBtn">
             <property name="styleSheet">
              <string notr="true">QPushButton {
                background-color: #1f6390;
                border: none;
                color: white;
                padding: 8px 16px; /* Adjusted padding */
                text-align: center;
                text-decoration: none;
                font-size: 14px; /* Adjusted font size */
                border-radius: 15px; /* Rounded corners */
                outline: none; /* No outline */
            }
            
            QPushButton:hover {
                background-color: #45a049; /* Darker green on hover */
            }
            
            QPushButton:pressed {
                background-color: #1f6390; /* Dark green when pressed */
            }</string>
             </property>
             <property name="text">
              <string>Check String</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="nfaToDfaBtn">
             <property name="styleSheet">
              <string notr="true">QPushButton {
                background-color: #1f6390;
                border: none;
                color: white;
                padding: 8px 16px; /* Adjusted padding */
                text-align: center;
                text-decoration: none;
                font-size: 14px; /* Adjusted font size */
                border-radius: 15px; /* Rounded corners */
                outline: none; /* No outline */
            }
            
            QPushButton:hover {
                background-color: #45a049; /* Darker green on hover */
            }
            
            QPushButton:pressed {
                background-color: #1f6390; /* Dark green when pressed */
            }</string>
             </property>
             <property name="text">
              <string>NFA -&gt; DFA</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="minimizeDfaBtn">
             <property name="styleSheet">
              <string notr="true">QPushButton {
                background-color: #1f6390;
                border: none;
                color: white;
                padding: 8px 16px; /* Adjusted padding */
                text-align: center;
                text-decoration: none;
                font-size: 14px; /* Adjusted font size */
                border-radius: 15px; /* Rounded corners */
                outline: none; /* No outline */
            }
            
            QPushButton:hover {
                background-color: #45a049; /* Darker green on hover */
            }
            
            QPushButton:pressed {
                background-color: #1f6390; /* Dark green when pressed */
            }</string>
             </property>
             <property name="text">
              <string>MInimize DFA</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <spacer name="horizontalSpacer_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="1" column="2">
     <spacer name="verticalSpacer_6">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="3">
     <spacer name="horizontalSpacer_4">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>886</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuAutomata_Simulator">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuSetting">
    <property name="title">
     <string>Setting</string>
    </property>
   </widget>
   <addaction name="menuAutomata_Simulator"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuSetting"/>
  </widget>
  <action name="actionNew">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
